File: src/main/java/io/github/takenoha/towerdefense/tactical/TacticalCandidate.java
package io.github.takenoha.towerdefense.tactical;

import java.util.Objects;

/** One stable candidate slot shown before a defense starts. */
public record TacticalCandidate(int slot, TacticalBuildDefinition definition) {
    public TacticalCandidate {
        if (slot < 0 || slot > 2) {
            throw new IllegalArgumentException("candidate slot must be between 0 and 2");
        }
        Objects.requireNonNull(definition, "definition");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/tactical/TacticalCandidate.kt
package io.github.takenoha.towerdefense.tactical

import java.util.Objects
import kotlin.jvm.JvmRecord

/** One stable candidate slot shown before a defense starts. */
@JvmRecord
data class TacticalCandidate(
    val slot: Int,
    val definition: TacticalBuildDefinition,
) {
    init {
        if (slot < 0 || slot > 2) {
            throw IllegalArgumentException("candidate slot must be between 0 and 2")
        }
        Objects.requireNonNull(definition, "definition")
    }
}
